### ISSUE TITLE
steam updated ! logined session can't accetp offer

### ISSUE THREAD

**Issue by ChetWei** [bukson/steampy#300]
steam updated ! steam logined session，but can't accetp offer, return 403

**Comment by a1063815047** [comment]
Yeah that's a big problem！！

**Comment by leon4312L** [comment]
can't get the session.
is_session_alive = steam_client.is_session_alive()
print(str(is_session_alive))

 it returns False

**Comment by ChetWei** [comment]
Maybe, after login success, the session is invalidated!

**Comment by a1063815047** [comment]
> 那么问题出在哪里呢？

https://github.com/DoctorMcKay/node-steamcommunity/issues/322

You can refer to this author’s

**Comment by woctezuma** [comment]
Maybe related:
- https://github.com/JustArchiNET/ArchiSteamFarm/issues/3043#issuecomment-1770261365

**Comment by leon4312L** [comment]
> So what is the problem?

cann't accept the tradeoffer. whatever gift offer or outgoing offer.

**Comment by Dandelion101** [comment]
I am running into the same problem.
I can use sessions that were created yesterday, but I can't use sessions that were created today.

**Comment by skyfeeder** [comment]
This morning Valve removed the ability to authorize via /login/dologin .
![image](https://github.com/bukson/steampy/assets/71432238/fe41d3d4-f07e-4016-8806-f2b3908ff800)

**Comment by kivinblue1** [comment]
I can login, but since this morning:
steampy/client.py", line 347, in make_offer_with_url
    if response.get('needs_mobile_confirmation'):
       ^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'

**Comment by Faramey** [comment]
I can login via `/login/dologin`, but session died instantly:
`{'success': True, 'requires_twofactor': False, 'login_complete': True, 'transfer_urls': ['https://store.steampowered.com/login/transfer', 'https://help.steampowered.com/login/transfer'], 'transfer_parameters': {'steamid': 'xxxxxxxxx', 'token_secure': None, 'auth': 'xxxxxxxxxxx', 'remember_login': False}}`

**Comment by ChetWei** [comment]
Using the new login api，to resolve this problem，but it post data has been encrypted.

**Comment by sattarov1960** [comment]
Is there a solution for this library?

**Comment by Dandelion101** [comment]
> Is there a solution for this library?

not yet
I think there will be an update soon, or the solution will appear on this page

**Comment by ZangYUzhang** [comment]
> made a decision?

Manually log in to obtain cookies and use it to modify the cookie data in the request.session

**Comment by wolfovik** [comment]
> > made a decision?
> 
> Manually log in to obtain cookies and use it to modify the cookie data in the request.session

Simply using cookies from the browser won't work. The mechanism has changed, and an AccessToken needs to be generated for successful requests.

**Comment by ZangYUzhang** [comment]
> > > made a decision?
> > 
> > 
> > Manually log in to obtain cookies and use it to modify the cookie data in the request.session
> 
> Simply using cookies from the browser won't work. The mechanism has changed, and an AccessToken needs to be generated for successful requests.

At least I'm using cookies in my browser for now，Can function normally, including conducting phone confirmation， just a temporary solution

**Comment by NewAge747** [comment]
Hi, if you wanna, can try my fix, seems like worked.

P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear

https://dropmefiles.com/uwYwC

**Comment by a1063815047** [comment]
> 你好，如果你愿意，可以尝试我的修复，看起来很有效。
> 
> PS但是我已经为自己重写了这个库的一半以上......我不确定什么对你有用，但我认为含义会很清楚
> 
> https://dropmefiles.com/uwYwC
> 
> PS2 PS2 是的，我删除了很多功能例外）））但是，它是热修复的，对吗？XD


It seems to be very well written, my friend, but there seems to be one missing post.
from Settings import Accounts

Traceback (most recent call last):
  File "get_account_money.py", line 8, in <module>
    from steampy.client import SteamClient
  File "/usr/local/python3/lib/python3.8/site-packages/steampy/client.py", line 11, in <module>
    from steampy.confirmation import ConfirmationExecutor
  File "/usr/local/python3/lib/python3.8/site-packages/steampy/confirmation.py", line 11, in <module>
    from steampy.login import InvalidCredentials
  File "/usr/local/python3/lib/python3.8/site-packages/steampy/login.py", line 3, in <module>
    from Settings import Accounts
ModuleNotFoundError: No module named 'Settings'

**Comment by skyfeeder** [comment]
> > 你好，如果你愿意，可以尝试我的修复，看起来很有效。
> > PS但是我已经为自己重写了这个库的一半以上......我不确定什么对你有用，但我认为含义会很清楚
> > https://dropmefiles.com/uwYwC
> > PS2 PS2 是的，我删除了很多功能例外）））但是，它是热修复的，对吗？XD
> 
> It seems to be very well written, my friend, but there seems to be one missing post. from Settings import Accounts
> 
> Traceback (most recent call last): File "get_account_money.py", line 8, in from steampy.client import SteamClient File "/usr/local/python3/lib/python3.8/site-packages/steampy/client.py", line 11, in from steampy.confirmation import ConfirmationExecutor File "/usr/local/python3/lib/python3.8/site-packages/steampy/confirmation.py", line 11, in from steampy.login import InvalidCredentials File "/usr/local/python3/lib/python3.8/site-packages/steampy/login.py", line 3, in from Settings import Accounts ModuleNotFoundError: No module named 'Settings'

You can remove strings with Settings and Accounts, lib will be work)

**Comment by a1063815047** [comment]
> Hi, if you wanna, can try my fix, seems like worked.
> 
> P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> 
> https://dropmefiles.com/uwYwC
> 
> P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD

Yes, thank you very much. It works. This is so well written.

**Comment by skyfeeder** [comment]
> Hi, if you wanna, can try my fix, seems like worked.
> 
> P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> 
> https://dropmefiles.com/uwYwC
> 
> P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD

For those who decide to use this fix: from the file login.py remove `from Settings import Accounts` and ```
for acc in Accounts:
            if acc.login == self.username:
                acc.ses = self.session
```
If you have previously saved your session to a file, overwrite it.
Thanks to @NewAge747 <3

**Comment by hsadhasd** [comment]
Can someone post the key parts of the file. The file doesn't work for me

**Comment by ZangYUzhang** [comment]
> Hi, if you wanna, can try my fix, seems like worked.
> 
> P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> 
> https://dropmefiles.com/uwYwC
> 
> P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD

Need to unzip your password?

**Comment by wolfovik** [comment]
123

**Comment by pudjojotaro** [comment]
i tried this fix, but keep getting 
Failed to login into account: 'client_id'

> Hi, if you wanna, can try my fix, seems like worked.
> 
> P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> 
> https://dropmefiles.com/uwYwC
> 
> P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD

**Comment by wolfovik** [comment]
> Hi, if you wanna, can try my fix, seems like worked.
> 
> P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> 
> https://dropmefiles.com/uwYwC
> 
> P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD

I used your work to obtain RefreshToken and AccessToken. I also rewrote a large part of the library. It works excellently!
Спасибо, чувак!

**Comment by wolfovik** [comment]
> i tried this fix, but keep getting Failed to login into account: 'client_id' any suggestions?
> 
> > Hi, if you wanna, can try my fix, seems like worked.
> > P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> > https://dropmefiles.com/uwYwC
> > P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD

It seems there's an incorrect link somewhere in your request. Most likely, an incorrect login_response parameter is passed in _update_steam_guard. _send_login_request returns the wrong value. 
Check the sequence of requests and the accuracy of the Steam URLs.

**Comment by pudjojotaro** [comment]
> > i tried this fix, but keep getting Failed to login into account: 'client_id' any suggestions?
> > > Hi, if you wanna, can try my fix, seems like worked.
> > > P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> > > https://dropmefiles.com/uwYwC
> > > P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD
> 
> It seems there's an incorrect link somewhere in your request. Most likely, an incorrect login_response parameter is passed in _update_steam_guard. _send_login_request returns the wrong value. Check the sequence of requests and the accuracy of the Steam URLs.

вроде разобрался, проблема была в другом; я частично использовал код отсюда : http://webcache.googleusercontent.com/search?client=opera-gx&q=cache%3Agithub.com%2FYESW0RLD%2FBuyOrderBot&sourceid=opera&ie=UTF-8&oe=UTF-8
и вызывая класс вне функции, все ломалось, но спасибо за совет!

**Comment by Snesik** [comment]
> Hi, if you wanna, can try my fix, seems like worked.
> 
> P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> 
> https://dropmefiles.com/uwYwC
> 
> P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD

Tell me, I'm trying to add your solution, but I get an error in the _pool_sessions_steam function response from the post request '{"response":{"had_remote_interaction":false}}'. Tell me what's the matter? Or at least where to look?

**Comment by spsdamn** [comment]
> Hi, if you wanna, can try my fix, seems like worked.
> 
> P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> 
> https://dropmefiles.com/uwYwC
> 
> P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD

I replaced your files and now in the method _send_login_request i have response: {"response":{"interval":5,"extended_error_message":""}} . Can you help?

**Comment by Snesik** [comment]
> > Hi, if you wanna, can try my fix, seems like worked.
> > P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> > https://dropmefiles.com/uwYwC
> > P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD
> 
> I replaced your files and now in the method _send_login_request i have response: {"response":{"interval":5,"extended_error_message":""}} . Can you help?

With a large number of requests, there seems to be a block for how long, try using another account. Or wait for a while

**Comment by Rasul11111** [comment]
> Hi, if you wanna, can try my fix, seems like worked.
> 
> P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> 
> https://dropmefiles.com/uwYwC
> 
> P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD

Спасибо

**Comment by spsdamn** [comment]
> > > Hi, if you wanna, can try my fix, seems like worked.
> > > P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> > > https://dropmefiles.com/uwYwC
> > > P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD
> > 
> > 
> > I replaced your files and now in the method _send_login_request i have response: {"response":{"interval":5,"extended_error_message":""}} . Can you help?
> 
> With a large number of requests, there seems to be a block for how long, try using another account. Or wait for a while

its true

**Comment by wwwscss** [comment]
Помогите нубу, что конкретно поменять в имеющиейся функции авторизации. Тг @yuki4onna. 
Заранее большое спасибо.

**Comment by Faramey** [comment]
> Hi, if you wanna, can try my fix, seems like worked.
> 
> P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> 
> https://dropmefiles.com/uwYwC
> 
> P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD

Thx a lot. It worked <3

**Comment by wwwscss** [comment]
Need help. 
   response = self.session.get(SteamUrl.GetPasswordRSAPublicKey_URL + self.username)
AttributeError: type object 'SteamUrl' has no attribute 'GetPasswordRSAPublicKey_URL'

**Comment by Faramey** [comment]
> Need help. response = self.session.get(SteamUrl.GetPasswordRSAPublicKey_URL + self.username) AttributeError: type object 'SteamUrl' has no attribute 'GetPasswordRSAPublicKey_URL'

Make sure you copy all endpoints from models.py

**Comment by bukson** [comment]
I dont have a time until next week or even further to manually fix it, but I will gladly and fast merge any pr fixing it, any volunteers :) ?

**Comment by Impressive57** [comment]
> https://dropmefiles.com/uwYwC

> For those who decide to use this fix: from the file login.py remove `from Settings import Accounts` and
> for acc in Accounts:
>             if acc.login == self.username:
>                 acc.ses = self.session

I used this solution and got the following error:

**fatal steam login error: CookieConflictError("There are multiple cookies with name, 'sessionid'")**

Any help, please? How to fix it?

**Comment by skyfeeder** [comment]
> > https://dropmefiles.com/uwYwC
> 
> > For those who decide to use this fix: from the file login.py remove `from Settings import Accounts` and
> > for acc in Accounts:
> > if acc.login == self.username:
> > acc.ses = self.session
> 
> I used this solution and got the following errors:
> 
> **fatal steam login error: CookieConflictError("There are multiple cookies with name, 'sessionid'")** **fatal steam login error: KeyError('client_id')**
> 
> Any help, please? How to fix it?

It looks like you are trying to load an old session from a file.

**Comment by leon4312L** [comment]
> Hi, if you wanna, can try my fix, seems like worked.
> 
> P.S. But I’ve already rewritten more than half of this library for myself... I’m not sure what will work for you, but I think the meaning will be clear
> 
> https://dropmefiles.com/uwYwC
> 
> P.S.2 P.S.2 Yes, I cut out a lot of feature exceptions ))) But, it hot fix, right? XD

much thanks. it works well.

**Comment by spsdamn** [comment]
After logining method is_session_alive return True. But request "SteamUrl.COMMUNITY_URL + '/tradeoffer/new/send" from make_offer respond with status_code 401. Am i alone with this problem?

**Comment by yesworId** [comment]
While using login() method 'Cannot perform redirects after login, no parameters fetched' issue appears. This is because of the new steam login update, right?

**Comment by wolfovik** [comment]
> I dont have a time until next week or even further to manually fix it, but I will gladly and fast merge any pr fixing it, any volunteers :) ?

Unfortunately, I'm busy this weekend. We'll need to neatly incorporate the fix into the release, as my version has too many differences. If no one beats me to it, I'll make a merge request at the beginning of the week.

**Comment by Impressive57** [comment]
Today Steam again give login-errors on some of my accounts. Some accounts work fine. What could this mean? Clean up saved cookies not help.
**fatal steam login error: KeyError('client_id')**

**Comment by Snesik** [comment]
> Today Steam again give login-errors on some of my accounts. Some accounts work fine. What could this mean? Clean up saved cookies not help. **fatal steam login error: KeyError('client_id')**

login_response = self._send_login_request() you need to see what's going on at this minute.

**Comment by booba-enjoyer** [comment]
```
    def _finallez_login(self):
        sessionid = self.session.cookies["sessionid"]
        redir = "https://steamcommunity.com/login/home/?goto="

        finallez_data = {
            'nonce': self.refresh_token,
            'sessionid': sessionid,
            'redir': redir
        }

        response = self.session.post("https://login.steampowered.com/jwt/finalizelogin", data=finallez_data)
        #print("FINALIZED", response.text)
        return response
```

Where do you get the sessionid cookie from? None of the responses from api.steampowered.com actually give you a cookie, do you generate your own sessionid?

**Comment by booba-enjoyer** [comment]
Just tested and it looks like the sessionid isn't necessary when sending that request

**Comment by NewAge747** [comment]
> Just tested and it looks like the sessionid isn't necessary when sending that request

    def _fetch_rsa_params(self, current_number_of_repetitions: int = 0) -> dict:
        maximal_number_of_repetitions = 5
        self.session.post(SteamUrl.COMMUNITY_URL)  # Получает первые куки
        #print("FIRST COOCKIE")
        response = self.session.get(St
 In login.py

**Comment by NewAge747** [comment]
> > > https://dropmefiles.com/uwYwC
> > 
> > 
> > > For those who decide to use this fix: from the file login.py remove `from Settings import Accounts` and
> > > for acc in Accounts:
> > > if acc.login == self.username:
> > > acc.ses = self.session
> > 
> > 
> > I used this solution and got the following errors:
> > **fatal steam login error: CookieConflictError("There are multiple cookies with name, 'sessionid'")** **fatal steam login error: KeyError('client_id')**
> > Any help, please? How to fix it?
> 
> It looks like you are trying to load an old session from a file.

Yeah, I forgot delete Accounts from file, in my program sessions are saved. Just delete this rows

**Comment by NewAge747** [comment]
I take https://github.com/MakcStudio/SteamAuth/ UserLogin.cs and rewrote on python, can check, maybe it will be more clear there. You need public LoginResult DoLoginV2()

**Comment by NewAge747** [comment]
> > Just tested and it looks like the sessionid isn't necessary when sending that request
> 
> ```
> def _fetch_rsa_params(self, current_number_of_repetitions: int = 0) -> dict:
>     maximal_number_of_repetitions = 5
>     self.session.post(SteamUrl.COMMUNITY_URL)  # Получает первые куки
>     #print("FIRST COOCKIE")
>     response = self.session.get(St
> ```
> 
> In login.py Just useless post, but take sessionid

{
                //Generate a SessionID
                SteamWeb.Request("https://steamcommunity.com/", "GET", proxy, proxy_type, "", cookies);
                SteamWeb.Request("https://store.steampowered.com/", "GET", proxy, proxy_type, "", cookies);
            }

            var c1 = cookies.GetCookies(new Uri("https://steamcommunity.com"));

            if (c1["sessionid"] == null || c1["sessionid"].Value == null)
                return LoginResult.GeneralFailure;
                
read next comment

**Comment by wolfovik** [comment]
#303 The update is ready. Ran the bot a bit, seems to work great! :)
Thank you, @NewAge747, for the tip.

**Comment by nbordin** [comment]
> #303 The update is ready. Ran the bot a bit, seems to work great! :) Thank you, @NewAge747, for the tip.

I had to change the for loop in set_sessionid_cookies with

```python
for name in ['steamLoginSecure', 'sessionid', 'steamRefresh_steam', 'steamCountry']:
    ....
```

to make it work.

@wolfovik thanks for the patch!

**Comment by wolfovik** [comment]
> > #303 The update is ready. Ran the bot a bit, seems to work great! :) Thank you, @NewAge747, for the tip.
> 
> I had to change the for loop in `set_sessionid_cookies` with
> 
> ```python
> for name in ['steamLoginSecure', 'sessionid', 'steamRefresh_steam', 'steamCountry']:
>     ....
> ```
> 
> to make it work.
> 
> @wolfovik thanks for the patch!


I didn't notice a difference, but I haven't tested everything thoroughly. I'll add your version, thank you!

**Comment by Fipsi742** [comment]
> > > #303 The update is ready. Ran the bot a bit, seems to work great! :) Thank you, @NewAge747, for the tip.
> > 
> > 
> > I had to change the for loop in `set_sessionid_cookies` with
> > ```
> > for name in ['steamLoginSecure', 'sessionid', 'steamRefresh_steam', 'steamCountry']:
> >     ....
> > ```
> > 
> > 
> > to make it work.
> > @wolfovik thanks for the patch!
> 
> I didn't notice a difference, but I haven't tested everything thoroughly. I'll add your version, thank you!

Did anyone write a fix for the create_buy_order command?
I managed to log in with @NewAge747 s hotfix (thank you :D) but couldnt make that command work so far.

Maybe i miss something but i only see fixes for the trading and not the market tools.

**Comment by johntanas** [comment]
I get {'response': {'had_remote_interaction': False}} for response = self._api_call('POST', 'IAuthenticationService', 'PollAuthSessionStatus', params = pool_data)

**Comment by bukson** [comment]
Fixed on master and in 1.1.0 realease

Thank you for all your support and changes, I've never suspected I will have a vibrant community around my little python project :)
